ledger-api/grpc-definitions/com/daml/ledger/api/v1/transaction_filter.proto

```diff
@@ -12,30 +12,60 @@ option java_outer_classname = "TransactionFilterOuterClass";
 option java_package = "com.daml.ledger.api.v1";
 option csharp_namespace = "Com.Daml.Ledger.Api.V1";
 
-// Used for filtering Transaction and Active Contract Set streams.
-// Determines which on-ledger events will be served to the client.
+// A filter both for filtering create and archive events as well as for
+// filtering transaction trees.
 message TransactionFilter {
 
-  // Keys of the map determine which parties' on-ledger transactions are being queried.
-  // Values of the map determine which events are disclosed in the stream per party.
-  // At the minimum, a party needs to set an empty Filters message to receive any events.
   // Each key must be a valid PartyIdString (as described in ``value.proto``).
+  // The interpretation of the filter depends on the stream being filtered:
+  // (1) For **transaction tree streams** only party filters with wildcards are allowed, and all subtrees
```

Comment:
currently it is not an error, if users specify InclusiveFilters. should it be then?
@stefanobaghino-da @meiersi-da @moritzkiefer-da

Reply:
Afaik, it is an error if you specify a template id in the filter on transaction tree streams. Are you suggesting that even specifying an empty InclusiveFilters should be an error? I don’t have particularly strong feelings there but it seems orthogonal to solving interface subscriptions so I wouldn’t change it now.

Reply:
I agree with orthogonality, this is only about the change of the documentation `are allowed` this somehow implies to me that if there are non-wildcard filters, than the query should be rejected. I am ok with not touching this aspect, but then probably we should change the wording. WDYT?

Reply:
I think I don’t fully understand what you’re getting at. To me a non-wildcard filter is something that specifies a template id (or multiple). And that _does_ get rejected at the moment for trees. So how are you suggesting we change the wording?

Reply:
Oh! Apologies! Of course you are right, I missed this. All good then :+1: